### DISCUSSION TITLE
When using an update query, it is still affected by the select policy.

### DISCUSSION THREAD

**Discussion by 2020-byte** [supabase/supabase#28774] [Questions] [answered]
<img width="797" alt="image" src="https://github.com/user-attachments/assets/242073cf-ec6a-42d1-9148-9474001ff5cb">
<img width="788" alt="image" src="https://github.com/user-attachments/assets/c5636004-a216-412f-930f-bb976a95d295">

```
const { error } = await supabase
          .from("people")
          .update({ is_deleted: true })
          .eq("id", person.id);
```

          
It is my understanding that Supabase now avoids performing a select after an update to bypass the select policy. However, when trying to change a value from is_deleted = false to is_deleted = true, I encounter the error: "new row violates row-level security policy for table." If I remove the is_deleted = false condition from the select policy, the update works fine, which suggests that the select policy is still affecting the update query. I would like to know how to resolve this issue.

Another strange issue is that if I remove the is_deleted = false condition from the select policy, the update query ignores the is_deleted = false condition specified in the update policy's using clause, and even records with is_deleted = true get updated to is_deleted = true.

**Comment by GaryAustin1** [comment, marked as answer]
Postgres requires updates meet select policy before and after.  This type of case comes up and rpc is likely the only way around it.
There are a couple of discussion on this here and I've seen some on the web about the same type "flag" with Postgres by itself.

https://www.postgresql.org/docs/current/sql-createpolicy.html
![image](https://github.com/user-attachments/assets/b2cf6a4d-02dd-4fcc-8571-db4ccc5156a5)

**Reply by 2020-byte** [reply]
Thank you for your clear and prompt response. I also appreciate the inclusion of the link!

I’m curious as to why it is possible to update a record from is_deleted=true to is_deleted=true when there is no select policy in place. The update policy’s using expression is supposed to check the condition that existed at the time the update query was executed. While this doesn’t hinder my intended implementation, I’m interested in understanding the reason behind this behavior.

**Reply by GaryAustin1** [reply]
Beyond my level of understanding about Postgres.  But my guess is that because they are permissive they are OR'd for the existing row so it gets thru based on the select policy.  

This seems like a good question for the Postgres community if it is not already asked and answered on the web.
